Kudu.Services/Deployment/DeploymentController.cs

```diff
@@ -451,7 +451,33 @@ public HttpResponseMessage GetResult(string id)
                 result.Url = baseUri;
                 result.LogUrl = UriHelper.MakeRelative(baseUri, "log");
 
-                return Request.CreateResponse(HttpStatusCode.OK, ArmUtils.AddEnvelopeOnArmRequest(result, Request));
+                HttpResponseMessage responseMessage;
+                if (ArmUtils.IsArmRequest(Request))
+                {
+                    switch (result.Status)
+                    {
+                        case DeployStatus.Building:
+                        case DeployStatus.Deploying:
+                        case DeployStatus.Pending:
+                            responseMessage = Request.CreateResponse(HttpStatusCode.Accepted, ArmUtils.AddEnvelopeOnArmRequest(result, Request));
+                            responseMessage.Headers.Location = Request.RequestUri;
+                            break;
+                        case DeployStatus.Failed:
+                            responseMessage = Request.CreateResponse(HttpStatusCode.BadRequest, ArmUtils.AddEnvelopeOnArmRequest(result, Request));
+                            break;
+                        case DeployStatus.Success:
+                            responseMessage = Request.CreateResponse(HttpStatusCode.OK, ArmUtils.AddEnvelopeOnArmRequest(result, Request));
+                            break;
+                        default:
+                            responseMessage = Request.CreateResponse(HttpStatusCode.BadRequest, ArmUtils.AddEnvelopeOnArmRequest(result, Request));
+                            break;
+                    }
+                }
```

Comment:
I'm thinking about this. It seems like it'll be a breaking change for any client also listing app specific deployments through ARM. 

Like I think the portal has a deployment list and drill down UI, that will start failing for failed deployments. I think maybe adding a `"provisioningState": "Failed"` property into the response might be a better option.

You could update the `DeployResult` object to have a `provisioningState` field that's only populated in ARM. And then instead of setting the response statusCode, set that filed to `Failed`, `InProgress` or `Succeeded`